Build instructions are inaccurate: a newer CMake is requried

# Issue

From the README:

> The required tooling on Linux is minimal — only a C++20 compiler, CMake ≥ 3.25, Ninja, and a handful of autotools packages

However, attempting to configure the build with CMake 3.28.3 results in an error:
```
CMake Error at CMakeLists.txt:7 (cmake_policy):
  Policy "CMP0156" is not known to this version of CMake.

[...]
-- Configuring incomplete, errors occurred!
```

# Steps to reproduce

Follow the build from source instructions at tag v26.6.5, using Ubuntu 24.04.4

# Workaround

During the attempt, a newer version of CMake is downloaded, so you can easily use that to try again:
```bash
$VCPKG_ROOT/downloads/tools/cmake-4.3.3-linux/cmake-4.3.3-linux-x86_64/bin/cmake --preset ci-linux
```

However, the build instructions should have the correct minimum version to start with.

--- a/README.md
+++ b/README.md
@@ -109,7 +109,7 @@ We support the following images in the Container Registry:
 ### Building from Source
 
 Dependencies are managed with [vcpkg](https://vcpkg.io/) in manifest mode.
-The required tooling on Linux is minimal — only a C++20 compiler, CMake ≥ 3.25,
+The required tooling on Linux is minimal — only a C++20 compiler, CMake ≥ 3.29,
 Ninja, and a handful of autotools packages that some vcpkg ports need to
 bootstrap themselves:
 

